Adding contribution related docs (#375)
primer/primitives: .github/pull_request_template.md

## Summary

<!--
A few sentences describing the changes being proposed in this pull request.
-->

## List of notable changes:

<!--
E.g.

> - **added** # for # component because #
> - **removed** props for # component because #
> - **updated** documentation for # component because #

Reviewer: ```suggestion
- **added** new design token for # because #
- **deprecated**  design token for # because #
- **updated** documentation for # because #
```

-->

-

## What should reviewers focus on?

-

## Steps to test:

<!--
Help reviewers test the feature by providing steps to reproduce the behavior.

> E.g.
> 
> 1. Open the # component in CI-deployed preview environment
> 2. Go to # story in Storybook
> 3. Verify that # behaves as described in the following issue.
> -->

Reviewer: ```suggestion
1. Open the preview documentation that has been deployed in this pull request
2. Go to # page
3. Verify that # behaves as described in the pull request description
```

1.
1.
1.

## Supporting resources (related issues, external links, etc):

-

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths

> - [ ] Check that there aren't other open Pull Requests for the same update/change

Reviewer: one more specific to this workstream...

```suggestion
- [ ] Check that there aren't other open Pull Requests for the same update/change
- [ ] Verify the design tokens changed in this PR are expected using the diffing results in this PR
```
